feat: add TestModeTransformation API documentation

michelboaventura opened the pull request:
Add comprehensive OpenAPI documentation for the test mode transformation API, which allows simulating state changes in shipments and requests during testing.

**Endpoints Added:**
- GET /test_mode_transformations/options - List available transformation types for a resource
- POST /test_mode_transformations - Create a new test mode transformation
- GET /test_mode_transformations - List transformations for a resource (paginated)
- GET /test_mode_transformations/:test_mode_transformation_id - Get specific transformation

**Features:**
- Full request/response schema definitions for TestModeTransformation resource
- Query parameters for filtering by resource ID and type (shipment/request)
- Documented transformation types: confirm, collect, transit, complete, cancel, expire
- Error responses for 400, 403, 404, and 422 status codes
- Pagination support for list endpoint
- Authentication via API key (test mode only)

dylanfareed commented on lib/docs_web/api_spec.ex:
```diff
@@ -1984,6 +1985,145 @@ Use the private url in the successful hosted session response to direct your use
+        },
+        "/test_mode_transformations/options" => %PathItem{
+          get: %Operation{
+            summary: "List Test Mode Transformation Types",
```
Lets update this to `List Test Mode Transformation Options` since that lines up better with the URL path and might be clearer that it returns available options for a specific resource.

michelboaventura replied:
Already applied — summary now reads `List Test Mode Transformation Options` (api_spec.ex:1991).

michelboaventura committed:
feat: add TestModeTransformation API documentation

Add comprehensive OpenAPI documentation for the test mode transformation API, which allows simulating state changes in
shipments and requests during testing.

Endpoints Added:

  *  GET /test_mode_transformations/options - List available transformation types for a resource
  *  POST /test_mode_transformations - Create a new test mode transformation
  *  GET /test_mode_transformations - List transformations for a resource (paginated)
  *  GET /test_mode_transformations/:test_mode_transformation_id - Get specific transformation

Features:

  * Full request/response schema definitions for TestModeTransformation resource
  * Query parameters for filtering by resource ID and type (shipment/request)
  * Documented transformation types: confirm, collect, transit, complete, cancel, expire
  * Error responses for 400, 403, 404, and 422 status codes
  * Pagination support for list endpoint
  * Authentication via API key (test mode only)

michelboaventura commented on lib/docs_web/schemas/request_body/test_mode_transformation_create.ex:
```diff
@@ -0,0 +1,47 @@
+              * `transit` - Updates a shipment to in-transit status
+              * `complete` - Marks a shipment as completed/delivered
+              * `cancel` - Cancels a shipment or request
+              * `expire` - Expires a request that is no longer needed",
```
Fixed in a3c29f7 — removed the stray `",` so the `expire` bullet no longer renders trailing quote/comma in the docs.

michelboaventura commented on lib/docs_web/api_spec.ex:
```diff
@@ -1984,6 +1985,145 @@ Use the private url in the successful hosted session response to direct your use
+              Retrieve the list of available test mode transformation types for a specific resource. Test mode transformations allow you to
+              simulate state changes in shipments and requests during testing.
+
+              This endpoint returns the transformation types that are valid for the specified resource based on its current state.
```
Already applied — description now reads "returns the transformations that are valid for the specified resource based on its current state" (api_spec.ex:1996).
